tests/bgp/test_bgpmon.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import logging
+from netaddr import IPAddress, IPNetwork
+import ptf.testutils as testutils
+from jinja2 import Template
+import ptf.packet as scapy
+from ptf.mask import Mask
+from tests.common.fixtures.ptfhost_utils import change_mac_addresses      # lgtm[py/unused-import]
+from tests.common.fixtures.ptfhost_utils import remove_ip_addresses       # lgtm[py/unused-import]
+from tests.common.helpers.generators import generate_ips as generate_ips
+
+pytestmark = [
+    pytest.mark.topology('any'),
+]
+
+BGPMON_TEMPLATE_FILE = 'bgp/bgpmon.j2'
+BGPMON_CONFIG_FILE = '/tmp/bgpmon.json'
+BGP_PORT = 179
+BGP_CONNECT_TIMEOUT = 120
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture
+def common_setup_teardown(duthost, ptfhost):
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    peer_addr = generate_ips(1, "%s/%s" % (mg_facts['minigraph_vlan_interfaces'][0]['addr'],
+                                           mg_facts['minigraph_vlan_interfaces'][0]['prefixlen']),
+                             [IPAddress(mg_facts['minigraph_vlan_interfaces'][0]['addr'])])[0]
+    peer_addr = str(IPNetwork(peer_addr).ip)
+    peer_port = mg_facts['minigraph_port_indices'][mg_facts['minigraph_vlans'][mg_facts['minigraph_vlan_interfaces'][0]['attachto']]['members'][0]]
+    local_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+    # Assign peer addr to an interface on ptf
+    ptfhost.shell("ifconfig eth{} {}".format(peer_port, peer_addr))
+    logger.info("Generated peer address {} and assigned to eth{} on ptf".format(peer_addr, peer_port))
+    bgpmon_args = {
+        'peer_addr': peer_addr,
+        'asn': mg_facts['minigraph_bgp_asn'],
+        'local_addr': local_addr,
+        'peer_name': 'bgp_monitor'
+    }
+    bgpmon_template = Template(open(BGPMON_TEMPLATE_FILE).read())
```

Comment:
better to use Ansible module `template` could do this.

Reply:
We need to add those extra vars to a ```AnsibleHost``` such as ```ptfhost``` or ```duthost``` if we want to use ```template``` in Ansible module, right?